storage/statedb/cache.go

```diff
@@ -16,39 +16,62 @@
 
 package statedb
 
-import "errors"
+import (
+	"errors"
+	"strings"
+)
+
+type TrieNodeCacheType string
+
+// TrieNodeCacheConfig contains configuration values of all TrieNodeCache.
+type TrieNodeCacheConfig struct {
+	CacheType          TrieNodeCacheType
+	FastCacheSizeMB    int      // Memory allowance (MB) to use for caching trie nodes in fast cache
+	RedisEndpoints     []string // Endpoints of redis cache
+	RedisClusterEnable bool     // Enable cluster-enabled mode of redis cache
+}
 
-type CacheType string
+// TrieNodeCache interface the cache of stateDB
+type TrieNodeCache interface {
+	Set(k, v []byte)
+	Get(k []byte) []byte
+	Has(k []byte) ([]byte, bool)
+}
 
 const (
-	// Available stateDB cache types
-	LocalCache  CacheType = "LocalCache"
-	RemoteCache           = "RemoteCache"
-	HybridCache           = "HybridCache"
+	// Available trie node cache types
+	CacheTypeFast   TrieNodeCacheType = "FastCache"
+	CacheTypeRedis                    = "RemoteCache"
+	CacheTypeHybrid                   = "HybridCache"
 )
 
 var (
-	errNotSupportedCacheType = errors.New("not supported stateDB Cache type")
+	ValidTrieNodeCacheTypes  = []TrieNodeCacheType{CacheTypeFast, CacheTypeRedis, CacheTypeHybrid}
```

Comment:
If you are not planning to use this somewhere else, how about defining this inside `ToValid`?
Since this is a slice, I am afraid this value is changed by other codes. Such situation occured in levelDB config in testcodes.

Reply:
The suggestion applied. Thanks!